src/time_machine/__init__.py

```diff
@@ -106,14 +125,34 @@ def extract_timestamp_tzname(
         elif dest.tzinfo == dt.timezone.utc:
             tzname = "UTC"
         elif dest.tzinfo is None:
-            dest = dest.replace(tzinfo=dt.timezone.utc)
+            if naive_mode == NaiveMode.MIXED or naive_mode == NaiveMode.UTC:
```

Comment:
```suggestion
            if naive_mode in {NaiveMode.MIXED, NaiveMode.UTC}:
```
This would improve readability and maintainability.